packages/button/src/LionButton.js

```diff
@@ -167,31 +171,13 @@ export class LionButton extends DisabledWithTabIndexMixin(SlotMixin(LitElement))
     }
   }
 
-  _redispatchClickEvent(oldEvent) {
-    // replacing `MouseEvent` with `oldEvent.constructor` breaks IE
-    const newEvent = new MouseEvent(oldEvent.type, oldEvent);
-    newEvent.__isRedispatchedOnNativeButton = true;
-    this.__enforceHostEventTarget(newEvent);
-    this._nativeButtonNode.dispatchEvent(newEvent);
-  }
-
   /**
-   * Prevent normal click and redispatch click on the native button unless already redispatched.
+   * Prevent normal click and redispatch submit on the native button unless already redispatched.
    */
```

Comment:
We also don't prevent normal click anymore, so that event delegation works as expected c.q. points to the right target...
And we dispatch a submit event (not re) to make it follow the exact same behaviour of a native button

Reply:
Yes, we should change the comment to something like "Dispatch submit event and invoke submit on the native form when clicked"